guardian/pluto-versions-manager
# Feature/azure compatibility

## What does this change?

Updates pluto-versions-manager to support Azure AD tokens

- token validation via remotely provided JWKS. This is cached for the duration of the pod's runtime.
  - jwks behaviour is enabled if the `tokenSigningCert` setting is an `http(s)` URL
  - otherwise the old, static key behaviour is used
- admin determination via the `roles` claim
  - if `roles` claim not present, then old behaviour is used
- update the frontend pluto-headers to the new version which supports azure AD
- frontend updates to support the changed functionality of some of the headers (requiring explicit contexts at the root level)

## How to test

- Run it up on dev. You should see no change - it should work as before against ADFS
- Run it up on local, pointing to Azure 
  - see https://github.com/guardian/pluto-headers/pull/8 for main configuration
  - set the `TOKEN_SIGNING_CERT` env var in prexit-local for versions-manager-deployment to the HTTPS url of the JWKS
  - ensure that `ADMIN_CLAIM_NAME` is correctly set to the name of the Azure role for administration
- You should be able to log in to Azure and have it work as normal as well

## How can we measure success?

Able to log in via either Azure or ADFS with only config changes

## Have we considered potential risks?

n/a so long as it's tested!
